src/pages/PaniniCreator.test.tsx

```diff
@@ -85,6 +111,115 @@ describe("Test form submitting", async () => {
+    });
+
+    test("It should reset all form configuration values when START AGAIN button is clicked", async () => {
+      // change values:
+      // swipe section (bread)
+      await userEvent.click(breadSetup.leftSwipeButton);
+      // multi swipe section (dressing)
+      await swipeElementsInteraction(
+        dressingSetup.leftSwipeButtons,
+        dressingVariants,
+        dressingSetup.swipeElementsDefaultValues
+      );
+      // select section (cheese, meat, egg)
+      await selectFieldInteraction(cheeseSetup.selectElements, cheeseVariants, cheeseSetup.selectDefaultValues);
+      await selectFieldInteraction(meatSetup.selectElements, meatVariants, meatSetup.selectDefaultValues);
+      await selectFieldInteraction(eggSetup.selectElements, eggVariants, eggSetup.selectDefaultValues);
+      // checkboxButton section (vegetables)
+      await checkboxButtonsInteraction(
+        vegetableSetup.checkboxButtonInteractionButtons,
+        vegetableVariant,
+        vegetableSetup.checkboxButtonsDefaultValues
+      );
+      // Checkbox section (spreads, topping, cutlery, napkins)
+      await checkboxButtonsInteraction(
+        spreadsSetup.checkboxInteractionButtons,
+        spreadVariant,
+        spreadsSetup.checkboxDefaultValues
+      );
+      await userEvent.click(toppingSetup.checkboxInteractionButton);
+      await userEvent.click(cutlerySetup.checkboxInteractionButton);
+      await userEvent.click(napkinsSetup.checkboxInteractionButton);
+      // Radio section (serving) field
+      await radioInteraction(servingSetup.radioInteractionButtons, servingSetup.radioDefaultValue);
+      // Text section (sandwichName field)
+      const notDefaultVal = "Not default name";
+      await userEvent.type(sandwichNameSetup.textInputElement, notDefaultVal);
+      // reset values:
+      const resetButton = screen.getByTestId("resetButton");
+      await userEvent.click(resetButton);
+      // check if values are default:
+      checkFormDefaultValues(
+        breadSetup,
+        dressingSetup,
+        cheeseSetup,
+        meatSetup,
+        eggSetup,
+        vegetableSetup,
+        spreadsSetup,
+        toppingSetup,
+        cutlerySetup,
+        napkinsSetup,
+        servingSetup,
+        sandwichNameSetup
+      );
+    });
+
+    test("Test panini form randomization", async () => {
+      const randomizeButton = screen.getByTestId("paniniRandomizeButton");
+      await userEvent.click(randomizeButton);
+      try {
+        checkFormDefaultValues(
+          breadSetup,
+          dressingSetup,
+          cheeseSetup,
+          meatSetup,
+          eggSetup,
+          vegetableSetup,
+          spreadsSetup,
+          toppingSetup,
+          cutlerySetup,
+          napkinsSetup,
+          servingSetup,
+          sandwichNameSetup
+        );
+      } catch (e) {
+        // task failed successfully, it shouldn't be equal to default values so if it throwed an error then it's randomized.
+        return;
```

Comment:
**nitpick:** What a dirty hack :D We should not use try catch in tests, as it will lead to unexpected situations, because you are dependent from error from other testing function. I think adding .eg.`expectFormValuesToNotBeDefault` would be a good direction